cognee/tasks/ingestion/migrate_relational_database.py

```
@@ -31,155 +36,231 @@ async def migrate_relational_database(graph_db, schema, migrate_column_data=True
+                    if not details["primary_key"]:
+                        primary_key_col = details["columns"][0]["name"]
+                    else:
+                        primary_key_col = details["primary_key"]
+
+                    # Query to find relationships based on foreign keys
+                    fk_query = text(
+                        f"SELECT {alias_1}.{primary_key_col} AS source_id, "
+                        f"{alias_2}.{fk['ref_column']} AS ref_value "
+                        f"FROM {table_name} AS {alias_1} "
+                        f"JOIN {fk['ref_table']} AS {alias_2} "
+                        f"ON {alias_1}.{fk['column']} = {alias_2}.{fk['ref_column']};"
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)